### PR TITLE
added cmake support

### PR DESCRIPTION
**Pull request by Arjun31415**
- [ ] Added an entry to `CHANGELOG.md` if this change could be valuable to users

Cmake treesitter and queries. 

image

### CODE REVIEW

File: lapce-core/Cargo.toml [hunk 1]
```diff
@@ -29,6 +29,7 @@ tree-sitter-bash = { git = "https://github.com/syntacti/tree-sitter-bash", branc
 tree-sitter-c = { version = "0.20.2", optional = true }
 tree-sitter-c-sharp = { git = "https://github.com/tree-sitter/tree-sitter-c-sharp", branch = "master", optional = true }
 tree-sitter-cpp = { version = "0.20.0", optional = true }
+tree-sitter-cmake = {version="0.1.0", optional = true}
```
Comment (panekj) [inline lapce-core/Cargo.toml:32]:
```suggestion
tree-sitter-cmake = { version="0.1.0", optional = true }
```

File: lapce-core/src/language.rs [hunk 2]
```diff
@@ -112,6 +112,8 @@ pub enum LapceLanguage {
     Bash,
     #[cfg(feature = "lang-c")]
     C,
+    #[cfg(feature = "lang-cmake")]
+    CMake,
```
Comment (panekj) [inline lapce-core/src/language.rs:116]:
```suggestion
    Cmake,
```

File: lapce-core/src/syntax/highlight.rs [hunk 3]
```diff
@@ -54,6 +54,7 @@ macro_rules! declare_language_highlights {
 declare_language_highlights!(
     Bash: "lang-bash",
     C: "lang-c",
+    CMake: "lang-cmake",
```
Comment (panekj) [inline lapce-core/src/syntax/highlight.rs:57]:
```suggestion
    Cmake: "lang-cmake",
```

File: lapce-core/src/language.rs [hunk 4]
```diff
@@ -245,6 +247,18 @@ const LANGUAGES: &[SyntaxProperties] = &[
         sticky_headers: &["function_definition", "struct_specifier"],
         extensions: &["c", "h"],
     },
+    #[cfg(feature = "lang-cmake")]
+    SyntaxProperties {
+        id: LapceLanguage::CMake,
```
Comment (panekj) [inline lapce-core/src/language.rs:252]:
```suggestion
        id: LapceLanguage::Cmake,
```

**Review by panekj — CHANGES_REQUESTED** [review]

File: lapce-core/src/syntax/highlight.rs (hunk 3 above)
Comment (Arjun31415) [reply lapce-core/src/syntax/highlight.rs:57]: The language id as provided in the crate has a capital M

Comment (panekj) [reply lapce-core/src/syntax/highlight.rs:57]: Doesn't matter, it's not dependent on external crate

**Comment by Arjun31415** [comment]
done as requested

File: lapce-core/Cargo.toml [hunk 5]
```diff
@@ -29,6 +29,7 @@ tree-sitter-bash = { git = "https://github.com/syntacti/tree-sitter-bash", branc
 tree-sitter-c = { version = "0.20.2", optional = true }
 tree-sitter-c-sharp = { git = "https://github.com/tree-sitter/tree-sitter-c-sharp", branch = "master", optional = true }
 tree-sitter-cpp = { version = "0.20.0", optional = true }
+tree-sitter-cmake = { version="0.1.0", optional = true }
```
Comment (panekj) [inline lapce-core/Cargo.toml:32]:
```suggestion
tree-sitter-cmake = { version = "0.1.0", optional = true }
```

**Review by panekj — APPROVED** [review]

**Pull request merged**